qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,346 @@
+        """Get the schedules from the Calibrations instance.
+
+        This method is called if :meth:`get_schedules_from_options` did not return
+        any schedules to use. Here, we get a schedule from an instance of
+        :class:`Calibrations` using the :meth:`get_schedule` method. Subclasses can override
+        this method if they need a different behaviour.
+
+        Args:
+            qubits: The qubits for which to fetch the schedules. If None is given this will
+                default to the physical qubits of the experiment.
+            sched_name: The name of the schedule to fetch from the calibrations. If None is
+                gven this will default to :code:`schedule_name` in the calibration options.
+            assign_params: A dict to specify parameters in the schedule that are
+                to be mapped to an unassigned parameter.
+
+        Returns:
+            A schedule for the corresponding arguments if there exists an instance
+            :code:`self.calibration_options.calibrations`.
+        """
+        cals = self.calibration_options.calibrations
+
+        if sched_name is None:
+            sched_name = self.calibration_options.schedule_name
+
+        if qubits is None:
+            qubits = self.physical_qubits
+
+        if cals is not None:
+            return cals.get_schedule(sched_name, qubits=qubits, assign_params=assign_params)
+
+        return None
+
+    def get_schedule_from_defaults(self, **kwargs) -> Optional[ScheduleBlock]:
+        """Get the schedules based on default experiment options.
+
+        Subclasses can override this method to define and get default schedules based on
+        default experiment options such as the number of samples in a Gaussian and its
+        amplitude. This function is called as a last resort in :meth:`get_schedules`
+        and accommodates cases when the user provides neither calibrations nor schedules.
+        For example, if the default schedule is a Gaussian then this function may return
+        the schedule
+
+        .. code-block:: python
+
+            with pulse.build(backend=backend, name="rabi") as default_schedule:
+                pulse.play(
+                    pulse.Gaussian(
+                        duration=self.experiment_options.duration,
+                        amp=Parameter("amp"),
+                        sigma=self.experiment_options.sigma,
+                    ),
+                    pulse.DriveChannel(self.physical_qubits[0]),
+            )
+
+        """
+        raise NotImplementedError(
+            f"{self.__class__.__name__} could not find a schedule in the experiment options "
+            "or the calibrations and no default schedule method was implemented."
+        )
+
+    def validate_schedule(self, schedule: ScheduleBlock):
+        """Subclass can implement this method to validate the schedule they use.
+
+        Validating schedules may include checks on the number of parameters and
```

Comment:
And as before, with all these functions having developer docs in API docs is a bit odd since subclasses will inherit them.